src/app/token-option-field-component-factories/spend-for-assignment-resubmission-token-option-field-component-factory.ts

```diff
@@ -81,20 +81,24 @@ export class SpendForAssignmentResubmissionTokenOptionFieldComponentFactory exte
                         return [
                             value,
                             [value.configuration.course.id, undefined],
-                            set(new Date(), {
-                                hours: 0,
-                                minutes: 0,
-                                seconds: 0,
-                                milliseconds: 0
-                            }),
+                            [
+                                set(new Date(), {
+                                    hours: 0,
+                                    minutes: 0,
+                                    seconds: 0,
+                                    milliseconds: 0
+                                }),
+                                value.configuration.course.timeZone
```

Comment:
We've identified a Canvas behavior that it does not allow "Available Until" or "Due At" time to be set between 00:00:00 and 00:00:59. According to [this comment](https://github.com/UCI-IN4MATX-191-Token-ATM/token-atm-spa/pull/64#issuecomment-1664935009), this behavior is confirmed to use the course time zone.

We decided to ignore the time zone inconsistency at that time. Since we are now adding the time zone support to Token ATM, we should also update the corresponding validation logic for this behavior, which is implemented at line `232-251` of `token-option-field-component-factory.ts`. It could be done by setting `value` to `utcToZonedTime(value, timeZone)` from `date-fns-tz` after line `233`. The `timeZone` argument could be either obtained from `DateTimeFieldComponent` by making `courseTimeZone` public or captured & stored via `StaticFormField`.

Reply:
The problem is that there is currently a validation applied to new due date field that prevents it being set to a time between "00:00:00" and "00:00:59" in the local timezone. For example, if the local timezone is UTC-8, and the course timezone is UTC-6. Now the instructor wants to configure the new due date to be 2 AM at UTC-6. To do so, the instructor needs to configure 00:00:00 in the new due date field since it uses the local timezone. Then the validation will prevent it being set, even though it is a totally valid configuration. On the other hand, if the instructor configure 22:00:00 in the field, the validation will pass, but Canvas will then automatically modify the date being set since it is 00:00:00 in the course timezone, which is not expected.

This issue already exists before as we assumes the course timezone and the local timezone being the same. The preview of course time just makes this issue more obvious and visible to the user.

Reply:
If you don't plan to add timezone support in this PR, I think we can leave this issue as is. But we should leave a reminder to fix this issue when we decide to add timezone support.